tests/unit/test_interpolation_tables.py

```diff
@@ -58,47 +59,155 @@ def diff(self, x, axis):
     low = np.random.rand(dim)
     high = 2 + np.random.rand(dim)
-    resolution = np.array([4 + d for d in range(dim)])
-    max_size = (high - low) / resolution
+    low = np.zeros(dim)
+    high = 2 * np.ones(dim)
+    # Vary the number of points in each dimension.
+    num_pt = np.array([4 + d for d in range(dim)])
+    # Grid resolution in each dimension
+    max_size = (high - low) / (num_pt - 1)
+    # Create function to be tested.
     function = factory(dim, max_size)
 
-    table = pp.InterpolationTable(low, high, resolution, function.func())
-
+    # We have occationally had issues with these tests failing on random data, however,
+    # subsequent testing (on huge arrays) failed to uncover any issues. Fix the random
+    # seed so that if errors arise, they should be reproducible.
+    # TODO: We should rather use hypothesis for this part.
+    np.random.seed(42)
+
+    # Create and test interpolation tables.
+    # Three types of tests are run:
+    #    1) The interpolated function should be evaluated exactly in the nodes of the
+    #       grid in parameter space.
+    #    2) The adaptive interpolation table can be built by adding points gradually
+    #       or all at once, the result should be the same.
+    #    3) Interpolation of values and derivatives have the expected accuracy relative
+    #       to the interpolated function in random points in parameter space.
+
+    # Create interpolation tables
+    table = pp.InterpolationTable(low, high, num_pt, function.func())
     adaptive_table = pp.AdaptiveInterpolationTable(
-        low, high, resolution, function.func()
+        dx=(high - low) / (num_pt - 1), base_point=low, function=function.func(), dim=1
+    )
+
+    # First check that the interpolation is exact in the nodes of the underlying grids.
+    # In this case, the interpolation weights in the tables should be calculated to be
+    # 1 in the relevant node, and 0 in adjacent nodes.
+    coords = np.vstack(table._coord)
+
+    adaptive_vals = adaptive_table.interpolate(coords)
+    table_vals = table.interpolate(coords)
+    known_vals = function.truth(coords)
+
+    # An error message from here likely indicates that either something is wrong with
+    # the function evaluation itself, or that the weights in the interpolation are
+    # wrong.
+    assert np.allclose(table_vals, adaptive_vals, atol=1e-10)
+    assert np.allclose(known_vals, adaptive_vals, atol=1e-10)
+
+    # Second test (only relevant for the adaptive table): Assign data to the table,
+    # instead of passing a function and letting the table compute the values. The two
+    # should be equivalent. We evaluate the table in the nodes in parameter space
+    # (e.g. variable coords), since the test is on assignment and not interpolation
+    # which is tested below.
+
+    # Assign all values at once.
+    adaptive_table_2 = pp.AdaptiveInterpolationTable(
+        dx=(high - low) / (num_pt - 1), base_point=low, dim=1
     )
 
-    npt = 100
```

Comment:
Despite `npt ` being fixed, consider parametrizing it.

Reply:
The number of points will vary between the dimensions (dimension 2 has one point more than dim 1 etc). That should be sufficient.